### ISSUE TITLE
Scrollbar in scrollbar

### ISSUE THREAD

**Issue by sefasenturk95** [zefoy/ngx-perfect-scrollbar#229]
I have a container that is scrollable and has the [perfectScrollbar] directive initialised on it. Within this container I have a form with a multiple select input. When you open this input you can scroll as well. When scrolling towards the bottom of the options in the select and checking a checkbox the container auto scrolls to the bottom of the screen.

Any ideas?

Example: https://www.dropbox.com/s/8gimqec28ws3562/example.mov?dl=0

**Comment by sconix** [comment]
Based on the video and your description it sounds something that should come from this library nor PS.  At least I can not think of anything that would cause that since there is no "automatic" action for scrolling at least in this library and afaik not in PS. So all scroll event needs to be requested. So my guess it that it needs to be something else. Have you tried leaving PS directive away and using native scroll?

**Comment by krojew** [comment]
I have also experienced something similar, but on touch scrolling. Scrolling an inner element also scrolls the parent, but only on touch gestures - mouse scrolling works just fine. Seems like touch events are being propagated to the parent.

**Comment by sconix** [comment]
Yes the PS does not have support for configurable touch propagation anymore. So that you need to control yourself. In the component there is an autoPropagation support that handles both touch and wheel event propagation automatically (i.e. only propagates them when scroll area is at the end).

**Comment by krojew** [comment]
I've already tried turning auto propagation off, and it didn't help.

**Comment by sconix** [comment]
You should turn it on. If does not help would like to know which browser / device. Its working in our app at least.

**Comment by krojew** [comment]
It's chrome on android. To reproduce, create a 100vh container with the scrollbar and a fixed height container inside with another one. Scrolling the inner one with touch also scrolls the outer one.

**Comment by sconix** [comment]
I will try to test this during this week, if you have time to put up a small reproduction in stackblitz it would speed things up.

**Comment by sconix** [comment]
I could not reproduce this so I would need to get a stackblitz example to see what you mean since I could be misinterpreting your description.

**Comment by krojew** [comment]
Here you go: https://stackblitz.com/edit/angular-mjoczb

On desktop - everything works. On mobile - touch scrolling the inner scrollable also scrolls the outer one.

**Comment by sconix** [comment]
In that example you are using directive so did you try the component and enabling autoPropagation as I instructed.

**Comment by krojew** [comment]
Enabling auto propagation seems to fix the issue. It's quite unintuitive - enabling auto propagation gives the effect of _not_ propagating touch events.

**Comment by sconix** [comment]
Well it means automatic propagation, so it enables or disables it depending on the state of the scrollbar. It disables propagation when not in the end of the scroll area and enables with small delay when the end of the scroll is reached.

**Comment by krojew** [comment]
The name suggests it simply automatically propagates the events, hence the confusion. It may be a good idea to add some explanation how it actually works.
